Sources/Orbit/Components/ListChoice.swift

```diff
@@ -0,0 +1,252 @@
+/// - Important: Component expands horizontally to infinity up to a ``Layout/readableMaxWidth``.
+public struct ListChoice: View {
+
+    let title: String
+    let description: String
+    let icon: Icon.Symbol
+    let trailingElement: ListChoice.TrailingElement
+    let backgroundColor: BackgroundColor?
+    let action: () -> Void
+
+    public var body: some View {
+        SwiftUI.Button(
+            action: {
+                HapticsProvider.sendHapticFeedback(.light(0.5))
+                action()
+            },
+            label: {
+                buttonContent
+            }
+        )
+        .buttonStyle(ListChoiceButtonStyle(backgroundColor: backgroundColor))
+        .accessibility(label: SwiftUI.Text(title))
+        .accessibility(hint: SwiftUI.Text(description))
+    }
+
+    @ViewBuilder var buttonContent: some View {
+        HStack(spacing: 0) {
+            header
+                .frame(maxWidth: .infinity, alignment: .leading)
+
+            // Provide minimal height (frame(minHeight:) collapses multiline text in snapshots)
+            Color.clear
+                .frame(width: 0, height: 48)
+
+            trailingElementView
+                .padding(.trailing, .medium)
+        }
+        .frame(maxWidth: Layout.readableMaxWidth, alignment: .leading)
+        .overlay(separator, alignment: .bottom)
+    }
+    
+    @ViewBuilder var header: some View {
+        if isHeaderEmpty == false {
+            HStack(alignment: .firstTextBaseline, spacing: .small) {
+                iconView
+                    .alignmentGuide(.firstTextBaseline) { _ in
+                        Heading.Style.title3.size * 1.1
+                    }
+
+                HStack(alignment: .firstTextBaseline, spacing: .xxSmall) {
+                    VStack(alignment: .leading, spacing: .xxxSmall) {
+                        Text(title)
+                        Text(description, size: .small, color: .inkLight)
+                    }
+                    .padding(.vertical, headerTextPadding)
+
+                    Spacer(minLength: 0)
+                }
+            }
+            .padding(.horizontal, .medium)
+        }
+    }
+
+    @ViewBuilder var trailingElementView: some View {
+        switch trailingElement {
+            case .none:
+                EmptyView()
+            case .disclosure:
+                Icon(symbol: .chevronRight, size: .medium, color: .inkNormal)
+            case .button(let type, let action):
+                switch type {
+                    case .add:
+                        Button(.plus, style: .primarySubtle, size: .small, action: action)
+                    case .remove:
+                        Button(.close, style: .criticalSubtle, size: .small, action: action)
+                }
+            case .checkbox(let isChecked, let isEnabled, let action):
+                Checkbox(state: isEnabled ? .normal : .disabled, isChecked: isChecked, action: action)
```

Comment:
The Checkbox here will be used as indicator only, meaning it should be `.disabled(true)`.
Similar with the `Button`s (it is not recommended to nest Buttons inside each other, it might not even work). 

The whole ListChoice component will probably have just a single `action` and the trailing elements will not be interactive on their own. (Similar behavior as in ChoiceTile, where radio/checkbox serve only as an non-interactive indicator).